node/src/chain/fallback.rs

```diff
@@ -99,15 +92,25 @@ impl<'a, N: Network, DB: database::DB, VM: vm::VMExecution>
             .await
             .get_provisioners(prev_header.state_hash)?;
 
-        let prev_provisioners_list = self
+        let mut provisioners_list = ContextProvisioners::new(provisioners_list);
+
+        let changed_provisioners = self
             .acc
             .vm
             .read()
             .await
-            .get_provisioners(prev_prev_header.state_hash)?;
-
-        let mut provisioners_list = ContextProvisioners::new(provisioners_list);
-        provisioners_list.set_previous(prev_provisioners_list);
+            .get_changed_provisioners(prev_header.state_hash)?;
+
+        if !changed_provisioners.is_empty() {
```

Comment:
nitpick: As this is duplicated code, can we put it into a function for easier maintenance?

Reply:
Done